### PR TITLE
[W5.3][F09-A4] Md Azhar

### PR DESCRIPTION
**Pull request by md-azsa**
Learning Outcomes - Inheritance
Pulled the commonalities from Address, Phone and Email methods to (new class) Contact

### CODE REVIEW

**Comment by eldon-chung** [comment]
Pull request looks good! Just wondering if maybe it should be okay to also add the isValid checks to the super class as well. Then in the super class definition we can raise an exception like how it's done the the Command class, to enfore that every subclass of Contact must implement their own isValid.

File: src/seedu/addressbook/data/person/Contact.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package seedu.addressbook.data.person;
+
+/**
+ * Learning outcome - Inheritance
+ * Extract commonalities from Address, Email, Phone
+ * Commonalities found: toString(), equals(Object x), hashCode(), isPrivate()
+ */
+public class Contact {
```
Comment (k-l-a) [inline src/seedu/addressbook/data/person/Contact.java:8]: This header comments does not conform to the coding standard.

File: src/seedu/addressbook/data/person/Contact.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    public String value;
```
Comment (k-l-a) [inline src/seedu/addressbook/data/person/Contact.java:10]: Could you have extracted more from the subclasses? Maybe their constructors?

**Review by k-l-a — COMMENTED** [review]
@md-azsa Some comments added. Please close PR after reading.

**Pull request closed without merging**